client/go/outline/reporting/reporting_client.go

```diff
@@ -0,0 +1,274 @@
+ * limitations under the License.
+ */
+
+package reporting
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"encoding/json"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/config"
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/connectivity"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func CheckTCPConnectivity(tcp transport.StreamDialer, reportingUrl string) error {
+	tcpErr := connectivity.CheckTCPConnectivityWithHTTP(tcp, reportingUrl)
+	if tcpErr != nil {
+		return tcpErr
+	}
+	return nil
+}
+
+const cookiesFile = "cookies.json"
+
+func Report(tcp transport.StreamDialer, reportingUrl string) (err error) {
+	// Perform a TCP connectivity check.
+	if err := CheckTCPConnectivity(tcp, reportingUrl); err != nil {
+		return fmt.Errorf("TCP connectivity check failed: %w", err)
+	}
+
+	// Load cookies from file
+	jar, err := cookiejar.New(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create cookie jar: %w", err)
+	}
+	var urls []*url.URL
+	urls, err = loadCookies(jar, cookiesFile)
+	if err != nil {
+		return fmt.Errorf("failed to load cookies: %w", err)
+	}
+
+	u, err := url.Parse(reportingUrl)
+	if err != nil {
+		return fmt.Errorf("failed to parse URL: %w", err)
+	}
+	cookies := jar.Cookies(u)
+
+	// Create a context with a timeout to avoid indefinite hangs.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Dial the stream using the StreamDialer.
+	conn, err := tcp.DialStream(ctx, reportingUrl)
+	if err != nil {
+		return fmt.Errorf("failed to connect to %s: %w", reportingUrl, err)
+	}
+	defer conn.Close()
+
+	// Create an HTTP POST request with cookies and a sample body
+	requestBody := "param1=value1&param2=value2"
+	request := "POST / HTTP/1.1\r\n" +
+		"Host: " + strings.TrimPrefix(reportingUrl, "https://") + "\r\n" +
+		"Content-Type: application/x-www-form-urlencoded\r\n" +
+		"Content-Length: " + fmt.Sprintf("%d", len(requestBody)) + "\r\n" +
+		"Connection: close\r\n"
+
+	for _, cookie := range cookies {
+		request += fmt.Sprintf("Cookie: %s=%s\r\n", cookie.Name, cookie.Value)
+	}
+	request += "\r\n" + requestBody
+
+	_, err = conn.Write([]byte(request))
+	if err != nil {
+		return fmt.Errorf("failed to write request: %w", err)
+	}
+
+	// Read the response from the server.
+	var response []byte
+	buffer := make([]byte, 4096)
+	for {
+		n, err := conn.Read(buffer)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return fmt.Errorf("failed to read response: %w", err)
+		}
+		response = append(response, buffer[:n]...)
+	}
+
+	// Parse the HTTP response to extract cookies.
+	headersEnd := bytes.Index(response, []byte("\r\n\r\n"))
+	if headersEnd == -1 {
+		return fmt.Errorf("failed to find end of headers in response")
+	}
+
+	headers := string(response[:headersEnd])
+	fmt.Println("HTTP Headers:")
+	fmt.Println(headers)
```

Comment:
Similar to above, debugging or logging? Please remove if it's for debugging, or use `slog` package if it's for logging.